lib/bucket.rb

```diff
@@ -0,0 +1,21 @@
+class Bucket
+
+  attr_reader :bucket_name, :client
+  
+  def initialize(name)
+    @client = Store.new.client
+    @bucket_name = name
+  end
+
+  def delete_bucket
+    begin
+      raise S3StoreArgumentError, 'S3StoreArgumentError :: Incorrect argument. bucket_name is mandatory!'.colorize(:red) if bucket_name.nil? || bucket_name.empty?
+      response = client.delete_bucket({bucket: bucket_name})
+      puts "Bucket #{bucket_name} deleted successfully!".colorize(:green) unless response.nil?
+    rescue Aws::S3::Errors::NoSuchBucket => e
+      puts "Aws::S3::Errors::NoSuchBucket :: #{bucket_name} does not exist!".colorize(:red)
+    rescue Aws::S3::Errors::PermanentRedirect => e
+      puts "Aws::S3::Errors::PermanentRedirect :: #{bucket_name} does not exist for your account!".colorize(:red)
```

Comment:
It appears that the message you are pushing out is the same? So what is the difference between `PermanentRedirect` and `NoSuchBucket`?

Reply:
While trying out this code snippet on the console, I encountered these two errors. I believe the redirect message maybe because there is another bucket with that name but not associated to my account. I am not sure entirely why it gave the redirect error. I used two bucket names which are not in my account. I ensured both names are valid name formats but just not part of my account. Once it gave the NoSuchBucket error and the other time it was the PermanentRedirect.